add a new flax example for Bert model inference (#34794)
huggingface/transformers: examples/flax/language-modeling/README.md

The input data are just random generated tokens, and the model is also jitted with JAX.
By default, it uses float32 precision for inference, users could use below commands to run inference with float32.
```bash
python3 run_bert_flax.py
```
> NOTE: For JAX Versions after v0.4.33 or later, users will need to set the below environment variables as a \
> temporary workaround to use Bfloat16 datatype. \
> This restriction is expected to be removed in future version
```bash
export XLA_FLAGS=--xla_cpu_use_thunk_runtime=false
```
Bfloat16 will give better performance on GPUs, and also Intel CPUs (Sapphire Rapids or later) with Advanced Matrix Extension (Intel AMX).  
By changing the dtype for FlaxBertModel to jax.numpy.bfloat16, users will get the performance benefits of underling hardware support.
on this Bert example.
```python
import jax
model = FlaxBertModel.from_pretrained("bert-base-uncased", config=config, dtype=jax.numpy.bfloat16)
```
To evaluate the performance speedup on bfloat16, users just need to add an argument "--precision bfloat16" for this example.

```bash
python3 run_bert_flax.py --precision bfloat16
```

> On a AWS c7i.4xlarge with Intel Sapphire Rapids, we get  > 2X speedup by changing precision from float32 to bfloat16.

Reviewer: ```suggestion
Switching from float32 to bfloat16 can increase the speed of an AWS c7i.4xlarge with Intel Sapphire Rapids by more than 2x.
```
